builtin/providers/cloudstack/resource_cloudstack_template.go

```diff
@@ -219,9 +236,16 @@ func resourceCloudStackTemplateRead(d *schema.ResourceData, meta interface{}) er
 	d.Set("password_enabled", t.Passwordenabled)
 	d.Set("is_ready", t.Isready)
 
-	setValueOrUUID(d, "os_type", t.Ostypename, t.Ostypeid)
-	setValueOrUUID(d, "zone", t.Zonename, t.Zoneid)
+	setValueOrUUID(d, "project", t.Project, t.Projectid)
 
+	if t.Zoneid == UnlimitedResourceID {
+		setValueOrUUID(d, "zone", t.Zonename, UnlimitedResourceID)
+	} else {
+		setValueOrUUID(d, "zone", t.Zonename, t.Zoneid)
+	}
```

Comment:
I don't see the difference here. First you check if `t.Zoneid == UnlimitedResourceID` and if so, you pass `UnlimitedResourceID` to `setValueOrUUID`.

But if `t.Zoneid` and `UnlimitedResourceID` are equal, you can just as well pass `t.Zoneid` right? So I guess this can be simplified to just:

```go
setValueOrUUID(d, "zone", t.Zonename, t.Zoneid)
```